### PR TITLE
Add cli command to query claimable rewards

### LINKED ISSUE #418: Add CLI command to query claimable rewards
It would be very helpful to query distribution, engagement and all rewards. For example:
* `tgrade query poe rewards  [address]` - all
* `tgrade query poe rewards  [address] --distribution` - query distribution contract only
* `tgrade query poe rewards [address] --engagement` - query engagement contract only
* `tgrade query poe rewards [address] --distribution --engagement` - query all

In order to safe fees, a compound script can run query before claim.

Relates to #242

### PR DESCRIPTION
**Pull request by pinosu**
Resolves #418

### CODE REVIEW

File: x/poe/client/cli/query.go [hunk 1]
```diff
@@ -375,19 +380,44 @@ $ %s query poe validator-reward %s1gghjut3ccd8ay0zduzj64hwre2fxs9ldmqhffj
...
+				}
+
+			}
+			return nil
 		},
 	}
 
 	flags.AddQueryFlagsToCmd(cmd)
+	cmd.Flags().BoolP(flagEngagement, "", false, "query engagement rewards")
+	cmd.Flags().BoolP(flagDistribution, "", false, "query distribution rewards")
```
Comment (alpe) [inline x/poe/client/cli/query.go:420]: defaults should be true to query the sum of both

File: x/poe/client/cli/query.go [hunk 2]
```diff
@@ -375,19 +380,44 @@ $ %s query poe validator-reward %s1gghjut3ccd8ay0zduzj64hwre2fxs9ldmqhffj
...
+				return err
 			}
-			res, err := queryClient.ValidatorOutstandingReward(context.Background(), req)
+			engRewards, err := cmd.Flags().GetBool(flagEngagement)
 			if err != nil {
 				return err
 			}
 
-			return clientCtx.PrintProto(res)
+			if distrRewards || !(distrRewards || engRewards) {
```
Comment (alpe) [inline x/poe/client/cli/query.go:392]: the "or" part does not look correct. Either distrRewards is set or not.

File: x/poe/client/cli/query.go [hunk 3]
```diff
@@ -375,19 +380,44 @@ $ %s query poe validator-reward %s1gghjut3ccd8ay0zduzj64hwre2fxs9ldmqhffj
...
+					return err
+				}
+
+				err = clientCtx.PrintProto(res)
+				if err != nil {
+					return err
+				}
+			}
+
+			if engRewards || !(distrRewards || engRewards) {
```
Comment (alpe) [inline x/poe/client/cli/query.go:407]: same as above

**Review by alpe — COMMENTED** [review]
This looks more complicated than it should be.
By making both params default to true, you can handle case 1 + 4.
Then it is just a like:
```
var total 
if distrRewards{
total += queryDistributionReward
}
if engRewards{
total += queryEngagementReward
}
print total

File: x/poe/client/cli/query.go [hunk 4]
```diff
@@ -375,22 +379,78 @@ $ %s query poe validator-reward %s1gghjut3ccd8ay0zduzj64hwre2fxs9ldmqhffj
...
 			}
-			res, err := queryClient.ValidatorOutstandingReward(context.Background(), req)
+			engRewards, err := cmd.Flags().GetBool(flagEngagement)
 			if err != nil {
 				return err
 			}
 
-			return clientCtx.PrintProto(res)
+			total := sdk.ZeroDec()
+			if !(distrRewards || engRewards) {
```
Comment (maurolacy) [inline x/poe/client/cli/query.go:392]: This would be the backwards compatible functionality. Nice.

**Review by maurolacy — APPROVED** [review]
LGTM. Let's wait for @alpe's review.

**Review by alpe — APPROVED** [review]
Very nice work!

**Pull request merged**